src/Unit-3/lesson2/README.md

```diff
@@ -0,0 +1,172 @@
+    {
+        Console.WriteLine($"{info.Key}: {info.Value}");
+    }
+}
+````
+The `Head` method expose a low level way to check if a potentialy large document
+exists, without loading it.
+
+## Exercise: Adding a order's line into a order document without loading the entire document
+Consider the document `orders/816` from the Northwind database.
+
+````csharp
+{
+    "Company": "companies/37",
+    "Employee": "employees/3",
+    "OrderedAt": "1998-04-30T00:00:00.0000000",
+    "RequireAt": "1998-05-28T00:00:00.0000000",
+    "ShippedAt": "1998-05-06T00:00:00.0000000",
+    "ShipTo": {
+        "Line1": "8 Johnstown Road",
+        "Line2": null,
+        "City": "Cork",
+        "Region": "Co. Cork",
+        "PostalCode": null,
+        "Country": "Ireland"
+    },
+    "ShipVia": "shippers/2",
+    "Freight": 81.73,
+    "Lines": [
+        {
+            "Product": "products/34",
+            "ProductName": "Sasquatch Ale",
+            "PricePerUnit": 14,
+            "Quantity": 30,
+            "Discount": 0
+        },
+        {
+            "Product": "products/40",
+            "ProductName": "Boston Crab Meat",
+            "PricePerUnit": 18.4,
+            "Quantity": 40,
+            "Discount": 0.1
+        },
+        {
+            "Product": "products/41",
+            "ProductName": "Jack's New England Clam Chowder",
+            "PricePerUnit": 9.65,
+            "Quantity": 30,
+            "Discount": 0.1
+        }
+    ]
+}
+```` 
+
+What if you want to add a order's line? Would we need to load the entire document? No!
+
+### Step 1: Create a new project and install the latest `RavenDB.Client` package
+
+Start Visual Studio and create a new `Console Application Project` named
+`UsingCommands`. Then, in the `Package Manager Console`, issue the following 
+command: 
+
+```Install-Package RavenDB.Client```
+
+### Step 2: Initialize the `DocumentStore` 
+
+Here we go again. let's manage the `DocumentStore` using our great friend `DocumentStoreHolder` pattern.  
+
+````csharp
+using System;
+using Raven.Client;
+using Raven.Client.Document;
+using Raven.Client.Indexes;
+
+namespace UsingCommands
+{
+    public static class DocumentStoreHolder
+    {
+        private static readonly Lazy<IDocumentStore> LazyStore =
+            new Lazy<IDocumentStore>(() =>
+            {
+                var store = new DocumentStore
+                {
+                    Url = "http://localhost:8080",
+                    DefaultDatabase = "Northwind"
+                };
+
+                store.Initialize();
+
+                return store;
+            });
+
+        public static IDocumentStore Store =>
+            LazyStore.Value;
+    }
+}
+````
+
+Remember to have started the RavenDB server.
+
+### Step 3: Update the document using a PatchRequest
+
+It's easy to change the document. 
+
+````csharp
+static void Main()
+{
+    var commands = DocumentStoreHolder.Store.DatabaseCommands;
+
+    var requestToAddLine = new PatchRequest
```

Comment:
Better to use scripted index here, instead of PatchRequest.
That is what we want to push